chore(docs): document basic Map methods

Tenpi opened the pull request:
I don't see why the common methods from Map shouldn't be documented, even if you just say "Identical to `Map.get()`". 

Feel free to check for correctness.

tipakA commented:
Because just next to `Collection` there's big `Extends Map`.
If you want to document Map methods, for literally same reason we should document all methods on objects that extend something, like `Client` extends `BaseClient`, which extends `EventEmitter`. Why not document all those under `Client` as well?

Tenpi commented:
Consider that many Discord.js users have never touched code before or looked at documentation before prior to creating their own bot. Hiding the most common collection methods behind a tiny "extends Map" text that can easily be missed is very misleading for beginners.

Monbrey commented:
Our continual advise is that beginners should not start with Discord.js anyway.

The reason this is consistently not-done across all the docs is that it introduces a workload on us to keep the docs updated in line with JS changes. Any new addition or changes to the `Map` interface would then need to be documented by us also.

This would also apply to every other case where we extend core JS or Node objects, like EventEmitter as tipaka suggested.

Linking to MDN via `extends` is more efficient and more correct.

Tenpi commented:
> Our continual advise is that beginners should not start with Discord.js anyway.
> 
> The reason this is consistently not-done across all the docs is that it introduces a workload on us to keep the docs updated in line with JS changes. Any new addition or changes to the `Map` interface would then need to be documented by us also.
> 
> This would also apply to every other case where we extend core JS or Node objects, like EventEmitter as tipaka suggested.
> 
> Linking to MDN via `extends` is more efficient and more correct.

Sorry, I should have explained the purpose of these changes clearer. The point is not to document every single method from parent classes, it's to make the documentation for extremely **common** methods more accessible for beginners. The first line I added was `Identical to Map.X()`, with the corresponding link to the MDN documentation. I can remove all of the documentation that I added if you would like, although it would look uglier in my opinion. But it would eliminate the concern of needing to keep the documentation up-to-date with MDN's. Lastly, if there is a breaking change in JS or Node objects, I think that updating the documentation will be of your least concern.

Monbrey commented:
Fair enough, I do actually agree with what you want to achieve here by documenting the CRUD style operators. The question does come up, "Why isn't X documented?" But where do we draw the line on what's "common"?

Some examples of what I mean:

`Map.values()` is used in documented examples on how to iterate a Collection in a for...of loop
`Map.forEach()` is commonly used also

Then as we keep going, it eventually gets to a point where it's like we're selectively _excluding_ certain methods.

General open discussion question I guess

Tenpi commented:
> Fair enough, I do actually agree with what you want to achieve here by documenting the CRUD style operators. The question does come up, "Why isn't X documented?" But where do we draw the line on what's "common"?
> 
> Some examples of what I mean:
> 
> `Map.values()` is used in documented examples on how to iterate a Collection in a for...of loop
> `Map.forEach()` is commonly used also
> 
> Then as we keep going, it eventually gets to a point where it's like we're selectively _excluding_ certain methods.
> 
> General open discussion question I guess

Thanks for agreeing. The CRUD methods are common, as in you can't really work with collections without using them. On second thought, I think a better word to use would be basic. I don't think that it's necessary to address more advanced methods in the documentation, since they aren't fundamental to working with collections.
